test/RequestClientWrapper.spec.ts

```diff
@@ -579,5 +580,142 @@ Code: 0 Message: You didnt pass any body`;
         done();
       });
     });
+
+    describe('Which has a maxConcurrentConnections config of 1', () => {
+      it('When making a single request', (done) => {
+        const TIMEOUT = 100;
+        sandbox
+          .stub(axios, 'get')
+          .callsFake(() => {
+            return new Promise((resolve) => {
+              setTimeout(resolve, 100);
+            });
+          });
+
+        requestClientWrapper = buildRequestClientWrapper(1);
+
+        const requestStartTime = new Date().getTime();
+        requestClientWrapper.get('/v1/orders/1', {}, (error, response, body) => {
+          const requestCompletionTime = new Date().getTime();
+          const requestTotalTime = requestCompletionTime - requestStartTime;
+          expect(requestTotalTime).to.be.lessThan(TIMEOUT + 25);
+          done();
+        });
+      });
+
+      it('When making 2 requests at the same time', (done) => {
+        const TIMEOUT = 100;
+        sandbox
+          .stub(axios, 'get')
+          .callsFake(() => {
+            return new Promise((resolve) => {
+              setTimeout(resolve, 100);
```

Comment:
I'm not a big fan of the time based unit tests but I understand that this is difficult to unit test. Another option could have been exposing `requestQueue` and `pendingRequests` instead of keeping them private so that this unit test could

Reply:
Please spend a little more time here to see if you can remove the time based aspect of these tests since they could break depending on what machine they are running on. If not we can go up with what you have but I won't be surprised if we need to refactor this later.

Reply:
Agreed @rjdavis3 @mb-ca . This is actually failing on my machine and the test is failing silently since done is never called properly. Only found it through logging:

`{ Error:  /v1/orders/1
  Status: 0
  Response Body:
  expected 162 to be below 125
    at requestPromise.then.catch (c:\Development\channelape-typescript-sdk\src\RequestClientWrapper.ts:158:34)
    at <anonymous>
  apiErrors: [],
  responseStatusCode: -1,
  responseStatusMessage: 'There was an error with the API' }`